client/index.ts

```diff
@@ -0,0 +1,452 @@
+        let cellForDeleteButton = tr.insertCell(-1);
+        let deleteButton = document.createElement('button');
+        deleteButton.type = 'button';
+        deleteButton.innerHTML = 'Delete';
+        cellForDeleteButton.appendChild(deleteButton);
+        deleteButton.addEventListener('click', function () {
+            let deleteObject = getCurrentRowData(i);
+            //Finding index of  object to be deleted in items Array
+            let index = -1;
+            for (let i = 0; i < crudObject.items.length; i++) {
+                if (crudObject.items[i].id === id) {
+                    index = i;
+                    break;
+                }
+            }
+            crudObject.delete(id, index, deleteObject);
+        });
+
+        //SAVE
+        let cellForSaveButton = tr.insertCell(-1);
+        let saveButton = document.createElement('button');
+        saveButton.type = 'button';
+        saveButton.innerHTML = 'Save';
+        cellForSaveButton.className = 'hiddenElements';
+        cellForSaveButton.id = 'saveButtonRow' + i;
+        cellForSaveButton.appendChild(saveButton);
+        cellForSaveButton.style.display = "none";
+        saveButton.addEventListener('click', function () {
+
+            let updatedRowObject = getCurrentRowData(i);
+
+            //Finding index of previous object in items Array
+            let index = -1;
+            for (let i = 0; i < crudObject.items.length; i++) {
+                if (crudObject.items[i].id === id) {
+                    index = i;
+                    break;
+                }
+            }
+
+            crudObject.update(id, index, updatedRowObject);
+
+            let saveButton = document.getElementById("saveButtonRow" + i)!;
+            let cancelButton = document.getElementById("cancelButtonRow" + i)!;
+            let headerSave = document.getElementById("headerSave")!;
+            let headerCancel = document.getElementById("headerCancel")!;
+
+            saveButton.style.display = "none";
+            cancelButton.style.display = "none";
+            headerSave.style.display = "none";
+            headerCancel.style.display = "none";
+        });
+
+        //CANCEL
+        let cellForCancelButton = tr.insertCell(-1);
+        let cancelButton = document.createElement('button');
+        cancelButton.type = 'button';
+        cancelButton.innerHTML = 'Cancel';
+        cellForCancelButton.className = 'hiddenElements';
+        cellForCancelButton.id = 'cancelButtonRow' + (i);
+        cellForCancelButton.appendChild(cancelButton);
+        cellForCancelButton.style.display = "none";
+        cancelButton.addEventListener('click', function () { showTable(); });
+
+
+    }
+
+    let divContainer = document.getElementById("showData")!;
+    divContainer.innerHTML = "";
+    divContainer.appendChild(table);
+    let loadButton = <HTMLButtonElement>document.getElementById("showDataButton")!;
+    loadButton.value = "Refresh";
+
+
+}
+
+
+function main() {
```

Comment:
main function does not denote anything. Provide proper and meaningful names to the functions and variables. also, add documentation to the functions.